### PR TITLE
auth bloc: logout

### PR DESCRIPTION
**Pull request by lantrannk**

### CODE REVIEW

File: lib/features/auth/screens/profile_screen.dart [hunk 1]
```diff
@@ -27,113 +29,118 @@ class _ProfileScreenState extends State<ProfileScreen> {
...
     final i10n = S.of(context);
 
-    return FutureBuilder(
-        future: UserStorage.getUser(),
-        builder: (_, snapshot) {
-          if (snapshot.hasData) {
-            user = snapshot.data!;
+    return BlocProvider(
+      create: (context) => AuthBloc(),
+      child: FutureBuilder(
```
Comment (nhandoanngoc-agilityio) [inline lib/features/auth/screens/profile_screen.dart]: When we apply bloc, we don't think still using FutureBuilder here.

File: lib/features/auth/screens/profile_screen.dart [hunk 2]
```diff
@@ -27,113 +29,118 @@ class _ProfileScreenState extends State<ProfileScreen> {
     late Map<String, dynamic> user;
     final i10n = S.of(context);
 
-    return FutureBuilder(
-        future: UserStorage.getUser(),
-        builder: (_, snapshot) {
-          if (snapshot.hasData) {
-            user = snapshot.data!;
+    return BlocProvider(
```
Comment (nhandoanngoc-agilityio) [inline lib/features/auth/screens/profile_screen.dart]: AuthBloc can provide on `main.dart` and we can use bloc by getting from context. If you still do not get my idea. Please create a call, I will explain in more detail

File: lib/core/widgets/indicator.dart [hunk 3]
```diff
@@ -22,3 +22,33 @@ class LoadingIndicator extends StatelessWidget {
     );
   }
 }
+
+class LoadingIndicator {
```
Comment (ningo-agilityio) [inline lib/core/widgets/indicator.dart:26]: Please separate a PR for this file

File: lib/features/auth/screens/profile_screen.dart [hunk 4]
```diff
@@ -27,113 +29,118 @@ class _ProfileScreenState extends State<ProfileScreen> {
...
+      create: (context) => AuthBloc(),
+      child: FutureBuilder(
+          future: UserStorage.getUser(),
+          builder: (_, snapshot) {
```
Comment (ningo-agilityio) [inline lib/features/auth/screens/profile_screen.dart]: Why do we need snapshot?

File: lib/features/auth/screens/profile_screen.dart [hunk 5]
```diff
@@ -27,113 +29,118 @@ class _ProfileScreenState extends State<ProfileScreen> {
...
+                                  onPressed: () {
+                                    AlertConfirmDialog.show(
+                                      context: context,
+                                      title: i10n.logoutConfirmTitle,
+                                      message: i10n.logoutConfirmMessage,
+                                      onPressedRight: () {
+                                        loadingIndicator.show(
+                                            context: context,
+                                            height: indicatorHeight);
+                                        BlocProvider.of<AuthBloc>(context).add(
```
Comment (ningo-agilityio) [inline lib/features/auth/screens/profile_screen.dart]: Using shorter syntax

File: lib/main.dart [hunk 6]
```diff
@@ -23,27 +25,39 @@ class MyApp extends StatelessWidget {
...
-    return MaterialApp(
-      title: 'Salon Appointment',
-      theme: themeData,
-      debugShowCheckedModeBanner: false,
-      localizationsDelegates: const [
-        S.delegate,
-        GlobalMaterialLocalizations.delegate,
-        GlobalWidgetsLocalizations.delegate,
-        GlobalCupertinoLocalizations.delegate,
+    return MultiBlocProvider(
```
Comment (nhandoanngoc-agilityio) [inline lib/main.dart]: We should use BlocProvider and create one bloc AuthBloc.

**Review by nhandoanngoc-agilityio — APPROVED** [review]

**Pull request merged**